### DISCUSSION TITLE
Move secret config one level up

### DISCUSSION THREAD

**Discussion by jagedn** [nextflow-io/nf-nomad#79] [Q&A] [answered]
Currently `secret` config is under `job` section

What do you think if we promote one level up in order to simplify the configuration ?

so instead

```
nomad {
    jobs {
        namespace = 'nf-nomad'
        secrets {
            enabled = true //if false then use LocalSecretsProvider implementation
        }
    }
}
```
use

```
nomad {

    client {
        address = "http://localhost:4646"
    }

    jobs {
        deleteOnCompletion = false
        volume = { type "host" name "scratchdir" }
        namespace = 'nf-nomad'
  }
   secrets {
          enabled = true //if false then use LocalSecretsProvider implementation
   }

}

```
@abhi18av ?

**Comment by abhi18av** [comment]
Hi @jagedn ,

I think it makes sense from the design perspective 👍 

Keeping in mind that the overall intent is that the specific secrets can also be provided via the `withName` or `withLabel` selectors in Nextflow config. 

Therefore, this `secrets.enabled` would be the only field within the `nomad.secrets` store no?

**Reply by jagedn** [reply]
no the only, also `path` is used

**Reply by abhi18av** [reply, marked as answer]
Ah, okay - found it https://github.com/nextflow-io/nf-nomad/blob/66dc75534ec7df63b0594b838844b400dcf549f9/plugins/nf-nomad/src/main/nextflow/nomad/config/NomadSecretOpts.groovy#L10 


Then, let's take this forward
